emperor/support_files/js/controller.js

```diff
@@ -1,37 +1,177 @@
-  this.sceneViews[3].resize(0.5 * this.width, 0.5 * this.height, 0.5 * this.width, 0.5 * this.height)
+
+  // the area we have to present the plot is smaller than the total
+  var plotWidth = this.$plotSpaceId.width();
+
+  if (this.sceneViews.length === 1) {
+    this.sceneViews[0].resize(0, 0, plotWidth, this.height);
+  }
+  else if (this.sceneViews.length === 2) {
+    this.sceneViews[0].resize(0, 0, 0.5 * plotWidth, this.height);
+    this.sceneViews[1].resize(0.5 * plotWidth, 0, 0.5 * plotWidth, this.height);
+  }
+  else if (this.sceneViews.length === 3) {
+    this.sceneViews[0].resize(0, 0, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[1].resize(0.5 * plotWidth, 0, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[2].resize(0, 0.5 * this.height, plotWidth, 0.5 * this.height);
+  }
+  else if (this.sceneViews.length === 4) {
+    this.sceneViews[0].resize(0, 0, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[1].resize(0.5 * plotWidth, 0, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[2].resize(0, 0.5 * this.height, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[3].resize(0.5 * plotWidth, 0.5 * this.height, 0.5 * plotWidth, 0.5 * this.height);
+  }
+  else {
+    throw Error('More than four views are currently not supported');
+  }
+
+  this.renderer.setSize(plotWidth, this.height);
+
+  // resize the grid according to the size of the container, since we are
+  // inside the tabs we have to account for that lost space, hence the 0.9
+  var gridWidth = this.$plotMenu.width() * 0.9,
+      gridHeight = this.$plotMenu.height() * 0.9;
+  $('#myGrid').width(gridWidth);
+  $('#myGrid').height(gridHeight);
 };
 
-EmperorController.prototype.render = function(){
+EmperorController.prototype.render = function() {
   this.renderer.setViewport(0, 0, this.width, this.height);
   this.renderer.clear();
   for (var i = 0; i < this.sceneViews.length; i++) {
     this.sceneViews[i].render();
+  }
+};
+
+EmperorController.prototype.buildUI = function() {
+
+  this.$plotMenu.append("<div id='emperor-menu-tabs'></div>");
+  $('#emperor-menu-tabs').append("<ul><li><a href='#keys'>Key</a></li></ul>");
+  $('#emperor-menu-tabs').append("<div id='keys' class='emperor-tab-div'></div>");
+  $('#emperor-menu-tabs').tabs({heightStyle: 'fill'});
+
+  var gridWidth = this.$plotMenu.width() * 0.9,
+      gridHeight = this.$plotMenu.height() * 0.9;
+
+  // http://stackoverflow.com/a/6602002
+  var $select = $("<select class='emperor-tab-drop-down'>");
+  _.each(this.dm.md_headers, function(header) {
+    $select.append($('<option>').attr('value', header).text(header));
+  });
+
+  $('#keys').append($select);
+  $('#keys').append("<div id='myGrid'></div>");
+  $('#myGrid').width(gridWidth);
+  $('#myGrid').height(gridHeight);
+
+  var grid;
+  var columns = [
+    {id: 'title', name: '', field: 'color', sortable: false,
+     maxWidth: 25, minWidth: 25, editor: ColorEditor,
+     formatter: ColorFormatter},
```

Comment:
Stupid question: is there any reason why `25` is chosen for `maxWidth` and `minWidth`?